### PR TITLE
TE-711: Implement the Promotion component

### PR DESCRIPTION
**Pull request by stephenkr**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-711)

### What **one** thing does this PR do?
Implements the Promotion component

### Any other notes
- CSS has been used to handle the hover blur effect and to displays the hover state button
- CSS has been used to overlay the image with a dark mask, this will make the text more readable

- The PR will also improve the styling for the Statistic component

__Preview__
kapture 2018-07-30 at 12 52 28

__Code Coverage__
screen shot 2018-07-30 at 11 45 50

### CODE REVIEW

File: src/components/general-widgets/Promotion/component.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          width={isDisplayedStacked ? 12 : 3}
+        >
+          <Statistic size="small">
+            <Statistic.Label>
+              <Paragraph weight="heavy">Save up to</Paragraph>
+            </Statistic.Label>
+            <Statistic.Value>
+              <Heading size="large">{discountAmount}</Heading>
+            </Statistic.Value>
+          </Statistic>
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: From the gif screengrab it looks like the statistic label and value are displaying in different greys. Please check in with Julio that it fits the mockup.

Comment (stephenkr) [reply src/components/general-widgets/Promotion/component.js]: Will do.

File: src/components/general-widgets/Promotion/Readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+```jsx
+<Promotion
+  backgroundImage='https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg'
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/Readme.md]: String props should be in double inverted commas.

Comment (stephenkr) [reply src/components/general-widgets/Promotion/Readme.md]: Fixing now.

File: src/components/general-widgets/Promotion/Readme.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  discountCode='F1KDH193S'
+  discountAmount='200€'
+  headingText='Summertime Festival'
+  onPromotionClick={console.log} />
+```
+
+### Variations
+
+#### Display as a card
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/Readme.md]: The description of the variation and the name of the prop should be related. 

`Display as a card` !== `isDiplayedStacked` 

Please choose one or the other and be consistent.

File: src/components/general-widgets/Promotion/component.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Statistic, Segment } from 'semantic-ui-react';
+import cx from 'classnames';
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: I think we use `getClassNames` in other components? Please be consistent.

File: src/components/general-widgets/Promotion/component.js (hunk 1 above)
Comment (stephenkr) [reply src/components/general-widgets/Promotion/component.js]: Actually i'll improve the text to match the design, the font color should be `#333` according to the design.

File: src/components/general-widgets/Promotion/component.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+          style={{ backgroundImage: `url(${backgroundImage})` }}
+          width={isDisplayedStacked ? 12 : 9}
+        >
+          <Grid padded>
+            <GridRow verticalAlign="top">
+              <GridColumn textAlign={isDisplayedStacked ? 'center' : 'left'}>
+                <Heading>{headingText}</Heading>
+              </GridColumn>
+            </GridRow>
+            {isDisplayedStacked ? (
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: Reduce noise with `{isDisplayedStacked && <GridRow>...`

File: src/components/general-widgets/Promotion/component.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+);
+
+Component.displayName = 'Promotion';
+
+Component.defaultProps = {
+  isDisplayedStacked: false,
+};
+
+Component.propTypes = {
+  /** The promotion background image */
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: No need for the word `promotion` here.

File: src/components/general-widgets/Promotion/component.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  backgroundImage: PropTypes.string.isRequired,
+  /** The displayed discount amount */
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: Inconsistency between lines 104 and 106 but they communicate the same thing. Please be careful to maintain consistency for ease of reading.

`The discount amount to be displayed`

File: src/components/general-widgets/Promotion/component.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+              <GridColumn textAlign={isDisplayedStacked ? 'center' : 'left'}>
+                <Heading>{headingText}</Heading>
+              </GridColumn>
+            </GridRow>
+            {isDisplayedStacked ? (
+              <React.Fragment />
+            ) : (
+              <GridRow className="book-now-button-container">
+                <GridColumn textAlign="center">
+                  <Button isRounded onClick={onPromotionClick}>
```
Comment (marcdavi-es) [inline src/components/general-widgets/Promotion/component.js]: Are you sure the click event should only be on the button, or on the whole component? Please check with Julio.

File: src/styles/semantic/themes/livingstone/elements/segment.overrides [hunk 9]
```diff
@@ -71,3 +71,117 @@
   height: 100vh;
   padding: 0;
 }
+
+/* Is Promotion */
```
Comment (marcdavi-es) [inline src/styles/semantic/themes/livingstone/elements/segment.overrides]: Please isolate all of the values in this sheet as variables in `.../segment.variables`

File: src/components/general-widgets/Promotion/Readme.md (hunk 3 above)
Comment (stephenkr) [reply src/components/general-widgets/Promotion/Readme.md]: Yep true, sorry i renamed this property and forgot to replace this, fixing now.

File: src/components/general-widgets/Promotion/component.js (hunk 4 above)
Comment (stephenkr) [reply src/components/general-widgets/Promotion/component.js]: Fixing now

File: src/components/general-widgets/Promotion/component.js (hunk 5 above)
Comment (stephenkr) [reply src/components/general-widgets/Promotion/component.js]: Ah good idea, fixing now.

File: src/components/general-widgets/Promotion/component.js (hunk 6 above)
Comment (stephenkr) [reply src/components/general-widgets/Promotion/component.js]: Ah ok, removing.

File: src/styles/semantic/themes/livingstone/elements/segment.overrides [hunk 10]
```diff
@@ -71,3 +71,122 @@
...
+  }
+
+  .content-section {
+    min-height: @promotionMinHeight;
+    position: relative;
+    overflow: hidden;
+
+    > * {
+      position: relative;
+      z-index: 3;
```
Comment (Jonnygwi) [inline src/styles/semantic/themes/livingstone/elements/segment.overrides]: These `z-index`s should be pulled out into variables and not just be arbitrary values. Semantic ui have a `z-index` map you could use.

File: src/styles/semantic/themes/livingstone/elements/segment.overrides [hunk 11]
```diff
@@ -71,3 +71,122 @@
...
+      background-image: inherit;
+      background-size: cover;
+      background-position: center;
+      z-index: 1;
+    }
+
+    &:after {
+      background-color: @black;
+      opacity: @promotionOverlayOpacity;
+      z-index: 2;
```
Comment (Jonnygwi) [inline src/styles/semantic/themes/livingstone/elements/segment.overrides]: Same here: https://github.com/lodgify/lodgify-ui/pull/257/files#r206174754

File: src/styles/semantic/themes/livingstone/elements/segment.overrides [hunk 12]
```diff
@@ -71,3 +71,122 @@
...
+      right: @promotionBackgroundOffset;
+      bottom: @promotionBackgroundOffset;
+      content: "";
+    }
+
+    &:before {
+      background-image: inherit;
+      background-size: cover;
+      background-position: center;
+      z-index: 1;
```
Comment (Jonnygwi) [inline src/styles/semantic/themes/livingstone/elements/segment.overrides]: Same here: https://github.com/lodgify/lodgify-ui/pull/257/files#r206174754

File: src/styles/semantic/themes/livingstone/elements/segment.overrides (hunk 10 above)
Comment (stephenkr) [reply src/styles/semantic/themes/livingstone/elements/segment.overrides]: Oh very good, I'll create some variables.

**Pull request merged**